File: Compiler/Code/bench/multi_lang_benchmark/w2_sieve/sieve.cpp
#include <iostream>
#include <vector>
#include <chrono>

int main() {
    auto t0 = std::chrono::high_resolution_clock::now();
    const int n = 100000;
    std::vector<int> flags(n + 1, 0);

    for (int p = 2; p < 320; ++p) {
        if (flags[p] == 0) {
            int mult = p * p;
            while (mult <= n) {
                flags[mult] = 1;
                mult += p;
            }
        }
    }

    int count = 0;
    for (int i = 2; i <= n; ++i) {
        if (flags[i] == 0) {
            count++;
        }
    }

    auto t1 = std::chrono::high_resolution_clock::now();
    auto us = std::chrono::duration_cast<std::chrono::microseconds>(t1 - t0).count();
    std::cout << "W2_SIEVE checksum=" << count << " time_us=" << us << "\n";
    return 0;
}

File: Compiler/Code/bench/multi_lang_benchmark/w2_sieve/sieve.rs
use std::time::Instant;

fn main() {
    let t0 = Instant::now();
    let n: usize = 100000;
    let mut flags = vec![0i32; n + 1];

    for p in 2..320 {
        if flags[p] == 0 {
            let mut mult = p * p;
            while mult <= n {
                flags[mult] = 1;
                mult += p;
            }
        }
    }

    let mut count = 0;
    for i in 2..=n {
        if flags[i] == 0 {
            count += 1;
        }
    }

    let us = t0.elapsed().as_micros();
    println!("W2_SIEVE checksum={} time_us={}", count, us);
}
